Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.cpp

```diff
@@ -278,28 +288,159 @@ namespace Terrain
+                {
+                    m_clipmapData.m_detailClipmapBoundsRegions[clipmapIndex + ClipmapConfiguration::DetailClipmapStackSizeMax * i] = zero;
+                }
+
+                Vector2i center = m_detailClipmapBounds[clipmapIndex].GetModCenter();
+                m_clipmapData.m_detailClipmapCenters[clipmapIndex][0] = center.m_x;
+                m_clipmapData.m_detailClipmapCenters[clipmapIndex][1] = center.m_y;
+                m_clipmapData.m_detailClipmapCenters[clipmapIndex][2] = center.m_x;
+                m_clipmapData.m_detailClipmapCenters[clipmapIndex][3] = center.m_y;
+            }
+
+            m_macroTotalDispatchThreadX = 1024;
+            m_macroTotalDispatchThreadY = 1024;
+            m_detailTotalDispatchThreadX = 1024;
+            m_detailTotalDispatchThreadY = 1024;
+
+            m_clipmapData.m_macroDispatchGroupCountX = m_macroTotalDispatchThreadX / MacroGroupThreadX;
+            m_clipmapData.m_macroDispatchGroupCountY = m_macroTotalDispatchThreadY / MacroGroupThreadY;
+
+            m_clipmapData.m_detailDispatchGroupCountX = m_detailTotalDispatchThreadX / DetailGroupThreadX;
+            m_clipmapData.m_detailDispatchGroupCountY = m_detailTotalDispatchThreadY / DetailGroupThreadY;
+
+            return;
+        }
+
         // macro clipmap data:
+        bool hasUpdate = false;
         for (uint32_t clipmapIndex = 0; clipmapIndex < m_macroClipmapStackSize; ++clipmapIndex)
         {
-            UpdateCenter(m_macroClipmapBounds[clipmapIndex], m_clipmapData.m_macroClipmapCenters[clipmapIndex]);
+            ClipmapBoundsRegionList updateRegionList = UpdateCenter(m_macroClipmapBounds[clipmapIndex], m_clipmapData.m_macroClipmapCenters[clipmapIndex]);
+
+            for (uint32_t i = 0; i < ClipmapBounds::MaxUpdateRegions; ++i)
+            {
+                if (updateRegionList.size() > i)
+                {
+                    AZStd::array<uint32_t, 4> aabb = { (uint32_t)updateRegionList[i].m_localAabb.m_min.m_x,
+                                                       (uint32_t)updateRegionList[i].m_localAabb.m_min.m_y,
+                                                       (uint32_t)updateRegionList[i].m_localAabb.m_max.m_x,
+                                                       (uint32_t)updateRegionList[i].m_localAabb.m_max.m_y };
+                    m_clipmapData.m_macroClipmapBoundsRegions[clipmapIndex + ClipmapConfiguration::MacroClipmapStackSizeMax * i] = aabb;
+                }
+                else
+                {
+                    m_clipmapData.m_macroClipmapBoundsRegions[clipmapIndex + ClipmapConfiguration::MacroClipmapStackSizeMax * i] = zero;
+                }
+            }
+
+            hasUpdate = hasUpdate || updateRegionList.size() > 0;
+        }
+
+        // [TODO] We need some more test to determine if dynamic dispatch number works better.
+        //        Or static numbers better than 64x64.
```

Comment:
I assume the TODO will get removed _very_ shortly in an upcoming PR?

Reply:
I need to run some performance tests. I'll remove this and create a ticket to track it.
And some more discussion is going on, we may change this part completely.